### PR TITLE
Make update-utmp not fail if it only fails to write wtmp records

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by mrc0mmand** [comment]
The CentOS CI fail is a yet another occurrence of #15978 - unrelated, *sigh*.

File: src/shared/utmp-wtmp.h [hunk 1]
```diff
@@ -24,6 +26,15 @@ int utmp_wall(
         bool (*match_tty)(const char *tty, void *userdata),
         void *userdata);
 
+static inline bool setutxentp(void) {
```
Comment (poettering) [inline src/shared/utmp-wtmp.h]: hmm, the "p" suffix was supposed to indicate that extra pointer level of indirection the destructors get, i.e. free() takes a pointer, but freep() gets a pointer to a pointer...

Doesn't matter much, but this naming logic doesn't really make sense in this case...

File: src/shared/utmp-wtmp.c [hunk 2]
```diff
@@ -109,19 +108,29 @@ static int write_entry_utmp(const struct utmpx *store) {
...
         utmpx = setutxentp();
 
-        if (!pututxline(store))
-                return -errno;
-        return 0;
+        if (pututxline(store))
+                return 0;
+        if (errno == ENOENT) /* If utmp/wtmp have been disabled, that's a good thing, hence
+                              * ignore the error. */
+                return 0;
```
Comment (poettering) [inline src/shared/utmp-wtmp.c:116]: maybe at least debug log?

File: src/shared/utmp-wtmp.c [hunk 3]
```diff
@@ -109,19 +108,29 @@ static int write_entry_utmp(const struct utmpx *store) {
...
 
         /* wtmp is a simple append-only file where each entry is
-        simply appended to the end; i.e. basically a log. */
+         * simply appended to the end; i.e. basically a log. */
 
         errno = 0;
         updwtmpx(_PATH_WTMPX, store);
+        if (errno == ENOENT) /* If utmp/wtmp have been disabled, that's a good thing, hence
+                              * ignore the error. */
+                return 0;
```
Comment (poettering) [inline src/shared/utmp-wtmp.c:132]: ditto

**Comment by keszybz** [comment]
I renamed setutxentp/endutxentp to utxent_start/utxent_cleanup and added the two requested log lines.

I also hit an interesting case: `sudo mv /run/utmp{,.bak}` resulted in an immediate logind crash and the graphical session going down :( But I don't think this is related to this PR. I'll try to investigate this later.

**Comment by mrc0mmand** [comment]
The CentOS CI fail is unrelated, see #16105...

**Pull request merged**